(at least) rust 1.54 nightly on dev

Over on https://github.com/conda-forge/maturin-feedstock/pull/32 it looks like we _might_ be blocked on `osx-arm64` by some features that are only in 1.54 nightly:

```
2021-08-15T17:47:51.2376740Z   {"reason":"compiler-message","package_id":"semver 1.0.3 (registry+https://github.com/rust-lang/crates.io-index)","manifest_path":"/Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/semver-1.0.3/Cargo.toml","target":{"kind":["lib"],"crate_types":["lib"],"name":"semver","src_path":"/Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/semver-1.0.3/src/lib.rs","edition":"2018","doc":true,"doctest":true,"test":true},"message":{"rendered":"error[E0658]: use of unstable library feature 'nonzero_leading_trailing_zeros'\n   --> /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/semver-1.0.3/src/identifier.rs:265:40\n    |\n265 |     let zero_bits_on_string_end = repr.leading_zeros();\n    |                                        ^^^^^^^^^^^^^\n    |\n    = note: see issue #79143 <https://github.com/rust-lang/rust/issues/79143> for more information\n    = help: add `#![feature(nonzero_leading_trailing_zeros)]` to the crate attributes to enable\n\n","children":[{"children":[],"code":null,"level":"note","message":"see issue #79143 <https://github.com/rust-lang/rust/issues/79143> for more information","rendered":null,"spans":[]},{"children":[],"code":null,"level":"help","message":"add `#![feature(nonzero_leading_trailing_zeros)]` to the crate attributes to enable","rendered":null,"spans":[]}],"code":{"code":"E0658","explanation":"An unstable feature was used.\n\nErroneous code example:\n\n```compile_fail,E0658\n#[repr(u128)] // error: use of unstable library feature 'repr128'\nenum Foo {\n    Bar(u64),\n}\n```\n\nIf you're using a stable or a beta version of rustc, you won't be able to use\nany unstable features. In order to do so, please switch to a nightly version of\nrustc (by using [rustup]).\n\nIf you're using a nightly version of rustc, just add the corresponding feature\nto be able to use it:\n\n```\n#![feature(repr128)]\n\n#[repr(u128)] // ok!\nenum Foo {\n    Bar(u64),\n}\n```\n\n[rustup]: https://rust-lang.github.io/rustup/concepts/channels.html\n"},"level":"error","message":"use of unstable library feature 'nonzero_leading_trailing_zeros'","spans":[{"byte_end":11391,"byte_start":11378,"column_end":53,"column_start":40,"expansion":null,"file_name":"/Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/semver-1.0.3/src/identifier.rs","is_primary":true,"label":null,"line_end":265,"line_start":265,"suggested_replacement":null,"suggestion_applicability":null,"text":[{"highlight_end":53,"highlight_start":40,"text":"    let zero_bits_on_string_end = repr.leading_zeros();"}]}]}}
2021-08-15T17:47:51.2394590Z   {"reason":"compiler-message","package_id":"semver 1.0.3 (registry+https://github.com/rust-lang/crates.io-index)","manifest_path":"/Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/semver-1.0.3/Cargo.toml","target":{"kind":["lib"],"crate_types":["lib"],"name":"semver","src_path":"/Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/semver-1.0.3/src/lib.rs","edition":"2018","doc":true,"doctest":true,"test":true},"message":{"rendered":"error[E0658]: use of unstable library feature 'nonzero_leading_trailing_zeros'\n   --> /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/semver-1.0.3/src/identifier.rs:354:37\n    |\n354 |     let len_bits = usize_bits - len.leading_zeros() as usize;\n    |                                     ^^^^^^^^^^^^^\n    |\n    = note: see issue #79143 <https://github.com/rust-lang/rust/issues/79143> for more information\n    = help: add `#![feature(nonzero_leading_trailing_zeros)]` to the crate attributes to enable\n\n","children":[{"children":[],"code":null,"level":"note","message":"see issue #79143 <https://github.com/rust-lang/rust/issues/79143> for more information","rendered":null,"spans":[]},{"children":[],"code":null,"level":"help","message":"add `#![feature(nonzero_leading_trailing_zeros)]` to the crate attributes to enable","rendered":null,"spans":[]}],"code":{"code":"E0658","explanation":"An unstable feature was used.\n\nErroneous code example:\n\n```compile_fail,E0658\n#[repr(u128)] // error: use of unstable library feature 'repr128'\nenum Foo {\n    Bar(u64),\n}\n```\n\nIf you're using a stable or a beta version of rustc, you won't be able to use\nany unstable features. In order to do so, please switch to a nightly version of\nrustc (by using [rustup]).\n\nIf you're using a nightly version of rustc, just add the corresponding feature\nto be able to use it:\n\n```\n#![feature(repr128)]\n\n#[repr(u128)] // ok!\nenum Foo {\n    Bar(u64),\n}\n```\n\n[rustup]: https://rust-lang.github.io/rustup/concepts/channels.html\n"},"level":"error","message":"use of unstable library feature 'nonzero_leading_trailing_zeros'","spans":[{"byte_end":15313,"byte_start":15300,"column_end":50,"column_start":37,"expansion":null,"file_name":"/Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/semver-1.0.3/src/identifier.rs","is_primary":true,"label":null,"line_end":354,"line_start":354,"suggested_replacement":null,"suggestion_applicability":null,"text":[{"highlight_end":50,"highlight_start":37,"text":"    let len_bits = usize_bits - len.leading_zeros() as usize;"}]}]}}
```

I have very degraded time in the next few weeks to dive into this, but wanted to capture this someplace more explicit.

--- a/.azure-pipelines/azure-pipelines-linux.yml
+++ b/.azure-pipelines/azure-pipelines-linux.yml
@@ -5,7 +5,7 @@
 jobs:
 - job: linux
   pool:
-    vmImage: ubuntu-16.04
+    vmImage: ubuntu-latest
   strategy:
     matrix:
       linux_64_:

--- a/.azure-pipelines/azure-pipelines-win.yml
+++ b/.azure-pipelines/azure-pipelines-win.yml
@@ -86,7 +86,7 @@ jobs:
 
     - script: |
         call activate base
-        conda.exe build "recipe" -m .ci_support\%CONFIG%.yaml
+        conda.exe build "recipe" -m .ci_support\%CONFIG%.yaml --suppress-variables
       displayName: Build recipe
       env:
         PYTHONUNBUFFERED: 1

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -1,7 +1,7 @@
-{% set version = "1.53.0" %}
+{% set version = "1.54.0" %}
 {% set year = "2021" %}
-{% set month = "03" %}
-{% set day = "24" %}
+{% set month = "08" %}
+{% set day = "16" %}
 
 package:
   name: rust-split
@@ -15,12 +15,12 @@ source:
   url: https://static.rust-lang.org/dist/{{ year }}-{{ month }}-{{ day }}/rust-nightly-x86_64-apple-darwin.tar.gz  # [osx and x86_64]
   url: https://static.rust-lang.org/dist/{{ year }}-{{ month }}-{{ day }}/rust-nightly-aarch64-apple-darwin.tar.gz  # [osx and arm64]
   url: https://static.rust-lang.org/dist/{{ year }}-{{ month }}-{{ day }}/rust-nightly-x86_64-pc-windows-msvc.tar.gz  # [win64]
-  sha256: 66df4113d438bcfacfae059eedd02546a61038de2c5baf8ea74e32e3f75e6e5c  # [win64]
-  sha256: 8a44b101f9467ef83dc412710501a3d6da2d939ba5a3772e8248515364218be6  # [osx and x86_64]
-  sha256: ac81df963e7dced9fdbb849f8312d56e2d11be35350ad83d01d5dd3a15f40191  # [linux and x86_64]
-  sha256: 2b3d12bfae92667eca2c3e230601c000766b3c772292b6d612f0ae3618de9f6f  # [linux and aarch64]
-  sha256: 84f6d5131167a643f1504126452da3f0d462850b647d33289907cc98cb242eef  # [linux and ppc64le]
-  sha256: 7e729e47dd66784eda98016ec2a33bcd217f2340dc1dc38ec79c4c8d4c40fb7b  # [osx and arm64]
+  sha256: 29e337c998149cb7f506ef3ebbbdd36a89885589cd27c54f2f31a4a844c857fc  # [win64]
+  sha256: b3607d66c68b2551dca122246a1bcbd47ec08dee5c46d1a60d41f356a1355b48  # [osx and x86_64]
+  sha256: db3e2bff208bd596a59ccc5dc86d0aff2051fd671b804c06f5973994374a021f  # [linux and x86_64]
+  sha256: 9f2f30f915ddac30054e587df3bb39b997c3857f2dd4547c60b5271a713bf257  # [linux and aarch64]
+  sha256: fb01bc7fb19d34b2a0ba1a26f8df80e2dc0f07e8d2d77d409f543ca207f39497  # [linux and ppc64le]
+  sha256: 6c662e8e441051ae5ddd186756457448ddf49a263382d677234bb8c273826da3  # [osx and arm64]
   patches:
     - 0001-gh-106-install.sh-Perfomance-Use-more-shell-builtins.diff
 

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -77,7 +77,10 @@ outputs:
         - set "SSL_CERT_FILE="       # [win]
         - unset REQUESTS_CA_BUNDLE  # [unix]
         - unset SSL_CERT_FILE       # [unix]
-        - rustc --version | grep "{{ version.replace('.', '\\.') }}.*{{ year }}-{{ month }}-{{ day }}"
+        - rustc --version
+        - rustc --version 1>version.txt 2>&1
+        # the day can be off-by-one
+        - cat version.txt | grep "rustc {{ version.replace('.', '\\.') }}-nightly \(.* {{ year }}-{{ month }}-.*\)"
         - rustc --help
         - rustdoc --help
         - cargo --help

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -1,4 +1,4 @@
-{% set version = "1.54.0" %}
+{% set version = "1.56.0" %}
 {% set year = "2021" %}
 {% set month = "08" %}
 {% set day = "16" %}
@@ -77,6 +77,7 @@ outputs:
         - set "SSL_CERT_FILE="       # [win]
         - unset REQUESTS_CA_BUNDLE  # [unix]
         - unset SSL_CERT_FILE       # [unix]
+        - rustc --version | grep "{{ version.replace('.', '\\.') }}.*{{ year }}-{{ month }}-{{ day }}"
         - rustc --help
         - rustdoc --help
         - cargo --help

